wandering-warriors/main.py

```diff
@@ -17,7 +18,136 @@ class TopMenu(Widget):
 
 
 class Abacus(Widget):
-    pass
+    MAX_BAR_W = 10
+    MIN_BORDER_W = 16
+    MAX_BEAD_SPACING = 8
+
+    def __init__(self, **kwargs):
+        super(Abacus, self).__init__(**kwargs)
+
+        self.n_bars = 12
+        self.n_top_beads = 1
+        self.n_bottom_beads = 4
+        self.bar_w = self.MAX_BAR_W
+
+        with self.canvas:
```

Comment:
Maybe move this entire context block into a function called `create_background`?